src/components/BusinessList/BusinessList.tsx

```diff
@@ -1,30 +1,30 @@
 import React from "react";
 import { Business } from "../../models/domain";
 import * as AR from "fp-ts/lib/Array";
-import { pipe } from "fp-ts/lib/function";
+import { pipe, flow } from "fp-ts/lib/function";
 import BusinessListItem from "../BusinessListItem/BusinessListItem";
-import { list, listItem } from "./BusinessList.treat";
+import { list as listStyle, listItem } from "./BusinessList.treat";
 import ErrorView from "../../design-system/ErrorView/ErrorView";
+import { nonEmptyArray, option } from "fp-ts";
 
 type Props = {
   businesses: Business[];
 };
 
 function BusinessList({ businesses }: Props) {
-  if (!businesses.length) {
-    return <ErrorView errorMessage="No element found :(" />;
-  }
-  return (
-    <ul className={list}>
-      {pipe(
-        businesses,
+  return pipe(
+    nonEmptyArray.fromArray(businesses),
+    option.fold(
+      () => <ErrorView errorMessage="No element found :(" />,
+      flow(
```

Comment:
actually i think you don't need to `pipe`: `fromArray` returns an `Option<NonEmptyArray<A>>` so you can `fold` directly on that

```
  return nonEmptyArray.fromArray(businesses).fold(
      () => <ErrorView errorMessage="No element found :(" />,
      nonEmptyBusinesses => ...
```

Reply:
If i try to `fold` directly on the result of `fromArray` i got an error: `Property 'fold' does not exist on type 'None'.` :(
`fromArray` returns an  `Option<NonEmptyArray<A>>` as you say but apparently it has no methods i can call on it.
![image](https://user-images.githubusercontent.com/3601911/89763994-e8734d00-daf3-11ea-8f68-efe8649539e0.png)
Am I missing something?

Reply:
Sorry, my bad, I just realized you are using `fp-ts v2`. In this case it is totally correct what you are doing 👍